### PR TITLE
Add homepage subfolder when generating packages for azure-sdk-for-javascript

### PR DESCRIPTION
**Pull request by daschult**
Resolves https://github.com/Azure/autorest.typescript/issues/182.

### CODE REVIEW

File: src/vanilla/Model/CodeModelTS.cs [hunk 1]
```diff
@@ -18,6 +18,11 @@ public class CodeModelTS : CodeModel
     {
         private const string ServiceClientOptions = "ServiceClientOptions";
 
+        private const string defaultGitHubRepositoryName = "azure-sdk-for-javascript";
```
Comment (RikkiGibson) [inline src/vanilla/Model/CodeModelTS.cs]: azure-sdk-for-js

File: src/vanilla/Model/CodeModelTS.cs [hunk 2]
```diff
@@ -115,6 +120,30 @@ public virtual IEnumerable<CompositeTypeTS> OrderedModelTemplateModels
...
+        public string HomePageUrl
+        {
+            get
+            {
+                string result = defaultGitHubUrl;
+                if (!string.IsNullOrEmpty(OutputFolder))
+                {
+                    string outputFolder = OutputFolder.Replace('\\', '/');
+                    int searchStringIndex = outputFolder.IndexOf(outputFolderSearchString, StringComparison.OrdinalIgnoreCase);
+                    if (0 <= searchStringIndex)
```
Comment (RikkiGibson) [inline src/vanilla/Model/CodeModelTS.cs]: yoda condition

File: src/vanilla/Model/CodeModelTS.cs (hunk 1 above)
Comment (daschult) [reply src/vanilla/Model/CodeModelTS.cs]: Done in https://github.com/Azure/autorest.typescript/pull/200/commits/f13760a76824c305882e4e0acb9a8085699a5666.

**Comment by RikkiGibson** [comment]
# 🤖 AutoRest automatic bundle size check 🤖
  ### Size increased by 0.33% (4.42 KB)

**Comment by RikkiGibson** [comment]
Size increase is just from testserver updates :)

**Pull request merged**